# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1298755262 18000
#      Sat Feb 26 16:21:02 2011 -0500
# Node ID 3244c6e0af4a237b002307c0b86b0abdddd1b221
# Parent  97d103fcb6ddeed0066df1eff6f095f8f66292fc
polynomial/polyval.m: Compute offset/normalization only when needed.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2011-02-26  Robert T. Short  <octave@phaselockedsystems.com>
+
+	* polynomial/polyval.m: Compute offset/normalization only when needed.
+
 2011-02-26  Robert T. Short <octave@phaselockedsystems.com>
 
 	* polynomial/polyval.m: Minor simplification in polynomial evaluation.
 
 2010-02-22  Rik  <octave@nomad.inbox5.com>
 
 	* general/bitcmp.m, general/bitget.m, general/bitset.m,
 	miscellaneous/license.m, special-matrix/pascal.m,
diff --git a/scripts/polynomial/polyval.m b/scripts/polynomial/polyval.m
--- a/scripts/polynomial/polyval.m
+++ b/scripts/polynomial/polyval.m
@@ -48,32 +48,31 @@ function [y, dy] = polyval (p, x, s, mu)
   if (nargin < 3)
     s = [];
   endif
 
   if (! (isvector (p) || isempty (p)))
     error ("polyval: first argument must be a vector");
   endif
 
-  if (nargin < 4)
-    mu = [0, 1];
+  if (nargin > 3)
+    x = (x - mu(1)) / mu(2);
   endif
 
   if (isempty (x))
     y = [];
     return;
   endif
 
   if (length (p) == 0)
     y = p;
     return;
   endif
 
   n = length (p) - 1;
-  x = (x - mu(1)) / mu(2);
   y = p(1) * ones (size (x));
   for i = 2:n+1
     y = y .* x + p(i);
   endfor
 
   if (nargout == 2)
     ## Note: the F-Distribution is generally considered to be single-sided.
     ## http://www.itl.nist.gov/div898/handbook/eda/section3/eda3673.htm
